src/actor/apply.rs

```diff
@@ -0,0 +1,155 @@
+use super::calc;
+use super::status_effect::status::{SetCombo, Status, StatusFlag, StatusFlags};
+use super::status_effect::StatusEffect;
+use super::{ActiveCombos, QueryActor};
+use crate::sim::{SimState, SimTime};
+use bevy_ecs::prelude::Entity;
+use std::sync::Arc;
+
+pub trait Apply {
+    fn apply(&self, sim: &SimState, query: &mut QueryActor, source: Entity, target: Entity);

```

Comment:
Decoupling "what should be done" from "actually doing the thing" using events is a very common and powerful pattern for breaking apart the monolithic state you see in `SimState` and `QueryActor`.